tests/server/routes/post-metrics.js

```diff
@@ -0,0 +1,475 @@
+          },
+
+          'gaCollector.write was called correctly': function () {
+            assert.strictEqual(mocks.gaCollector.write.callCount, 1);
+            var args = mocks.gaCollector.write.args[0];
+            assert.lengthOf(args, 1);
+            assert.equal(args[0], mocks.request.body);
+          },
+
+          'activityEvent was called correctly': function () {
+            assert.strictEqual(mocks.activityEvent.callCount, 1);
+            var args = mocks.activityEvent.args[0];
+            assert.lengthOf(args, 3);
+            assert.equal(args[0], 'flow.begin');
+            assert.isObject(args[1]);
+            assert.lengthOf(Object.keys(args[1]), 3);
+            assert.equal(args[1].flow_id, 'qux');
+            assert.strictEqual(args[1].flow_time, 0);
+            assert.equal(args[1].time, 42);
+            assert.equal(args[2], mocks.request);
+          }
+        }
+      },
+
+      'route.process without flow.begin event': {
+        setup: function () {
+          mocks.request = {
+            body: {
+              events: [
+                { type: 'foo' },
+                { type: 'bar' },
+                { type: 'baz' }
+              ],
+              flowBeginTime: 42,
+              flowId: 'qux',
+              isSampledUser: true
+            },
+            get: sinon.spy(function (header) {
+              switch (header.toLowerCase()) {
+              case 'content-type':
+                return 'application/json';
+              case 'user-agent':
+                return 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:47.0) Gecko/20100101 Firefox/47.0';
+              }
+              return '';
+            })
+          };
+          mocks.response = { json: sinon.spy() };
+          mocks.nextTick = sinon.spy();
+          var nextTickCopy = process.nextTick;
+          process.nextTick = mocks.nextTick;
+          instance.process(mocks.request, mocks.response);
+          process.nextTick = nextTickCopy;
+        },
+
+        'response.json was called': function () {
+          assert.equal(mocks.response.json.callCount, 1);
+        },
+
+        'process.nextTick was called': function () {
+          assert.equal(mocks.nextTick.callCount, 1);
+        },
+
+        'process.nextTick callback': {
+          setup: function () {
+            mocks.nextTick.args[0][0]();
+          },
+
+          'mozlog.error was not called': function () {
+            assert.strictEqual(mocks.mozlog.error.callCount, 0);
+          },
+
+          'metricsCollector.write was called': function () {
+            assert.strictEqual(mocks.metricsCollector.write.callCount, 2);
+          },
+
+          'statsdCollector.write was called': function () {
+            assert.strictEqual(mocks.statsdCollector.write.callCount, 2);
+          },
+
+          'gaCollector.write was called': function () {
+            assert.strictEqual(mocks.gaCollector.write.callCount, 2);
```

Comment:
Plus, most of the tests here are for stuff I haven't touched. The metrics route handler had no test coverage at all before this PR, so we're definitely in a better position tests-wise than we used to be.